frontend/test/pytest/test_preprocess.py

```diff
@@ -244,8 +249,387 @@ def circuit(theta: float):
+
+        with pytest.raises(
+            CompileError, match="Must use 'measure' from Catalyst instead of PennyLane."
+        ):
+            with EvaluationContext(EvaluationMode.QUANTUM_COMPILATION) as ctx:
+                _ = catalyst_decompose(tape, ctx, stopping_condition)
+
+    def test_error_for_pennylane_midmeasure_decompose_nested(self):
+        """Test that an error is raised in decompose if a PennyLane mid-circuit measurement
+        is encountered"""
+
+        stopping_condition = partial(catalyst_acceptance, operations=expected_ops)
+
+        with qml.queuing.AnnotatedQueue() as q:
+            qml.RX(1.23, wires=0)
+            qml.measure(0)
+
+        ops, measurements = qml.queuing.process_queue(q)
+        subtape = qml.tape.QuantumScript(ops, measurements)
+
+        region = HybridOpRegion([], subtape, [], [])
+        region.trace = None
+        adjoint_op = Adjoint([], [], [region])
+
+        tape = qml.tape.QuantumScript([adjoint_op, qml.Y(1)], [])
+
+        with pytest.raises(
+            CompileError, match="Must use 'measure' from Catalyst instead of PennyLane."
+        ):
+            with EvaluationContext(EvaluationMode.QUANTUM_COMPILATION) as ctx:
+                _ = catalyst_decompose(tape, ctx, stopping_condition)
+
+    def test_unsupported_op_with_no_decomposition_raises_error(self):
+        """Test that an unsupported operator that doesn't provide a decomposition
+        raises a CompileError"""
+
+        # operations=[], all ops are unsupported
+        stopping_condition = partial(catalyst_acceptance, operations=[])
+
+        tape = qml.tape.QuantumScript([qml.Y(0)])
+
+        with pytest.raises(
+            CompileError,
+            match="not supported with catalyst on this device and does not provide a decomposition",
+        ):
+            with EvaluationContext(EvaluationMode.QUANTUM_COMPILATION) as ctx:
+                _ = catalyst_decompose(tape, ctx, stopping_condition)
+
+    def test_decompose_to_matrix_raises_error(self):
+        """Test that _decompose_to_matrix raises a CompileError if the operator has no matrix"""
+
+        # operations=[], all ops are unsupported
+        stopping_condition = partial(catalyst_acceptance, operations=[])
+
+        class NoMatrixMultiControlledX(qml.MultiControlledX):
+            """A version of MulitControlledX with no matrix defined"""
+
+            def matrix(self):
+                """raise an error"""
+                raise qml.operation.MatrixUndefinedError
+
+        tape = qml.tape.QuantumScript([NoMatrixMultiControlledX(wires=[0, 1, 2, 3])])
+
+        with pytest.raises(CompileError, match="could not be decomposed, it might be unsupported"):
+            with EvaluationContext(EvaluationMode.QUANTUM_COMPILATION) as ctx:
+                _ = catalyst_decompose(tape, ctx, stopping_condition)
+
+
 class TestTransform:
-    """Test the transforms implemented in Catalyst."""
+    """Teented in Catalyst."""
```

Comment:
?

Reply:
Looks like I somehow deleted `st the transforms implem`? 🙈 

Will undo, thanks!